qiskit/providers/backend_compat.py

```diff
@@ -0,0 +1,249 @@
+    else:
+        target = Target(num_qubits=configuration.n_qubits)
+        for gate in configuration.gates:
+            name = gate.name
+            gate_props = (
+                {tuple(x): None for x in gate.coupling_map}  # type: ignore[misc]
+                if hasattr(gate, "coupling_map")
+                else {None: None}
+            )
+            gate_len = len(gate.coupling_map[0]) if hasattr(gate, "coupling_map") else 0
+            if name in name_mapping:
+                target.add_instruction(name_mapping[name], gate_props)
+            else:
+                custom_gate = Gate(name, gate_len, [])
+                target.add_instruction(custom_gate, gate_props)
+        target.add_instruction(Measure())
+    # parse global configuration properties
+    if hasattr(configuration, "dt"):
+        target.dt = configuration.dt
+    if hasattr(configuration, "timing_constraints"):
+        target.granularity = configuration.timing_constraints.get("granularity")
+        target.min_length = configuration.timing_constraints.get("min_length")
+        target.pulse_alignment = configuration.timing_constraints.get("pulse_alignment")
+        target.aquire_alignment = configuration.timing_constraints.get("acquire_alignment")
+    # If a pulse defaults exists use that as the source of truth
+    if defaults is not None:
+        inst_map = defaults.instruction_schedule_map
+        for inst in inst_map.instructions:
+            for qarg in inst_map.qubits_with_instruction(inst):
+                sched = inst_map.get(inst, qarg)
+                if inst in target:
+                    try:
+                        qarg = tuple(qarg)
+                    except TypeError:
+                        qarg = (qarg,)
+                    if inst == "measure":
+                        for qubit in qarg:
+                            target[inst][(qubit,)].calibration = sched
+                    elif qarg in target[inst]:
+                        target[inst][qarg].calibration = sched
+    if "delay" not in target:
+        target.add_instruction(
+            Delay(Parameter("t")), {(bit,): None for bit in range(target.num_qubits)}
+        )
+    return target
+
+
+def qubit_props_list_from_props(
+    properties: BackendProperties,
+) -> List[QubitProperties]:
+    """Uses BackendProperties to construct
+    and return a list of IBMQubitProperties.
+    """
+    qubit_props: List[QubitProperties] = []
+    for qubit, _ in enumerate(properties.qubits):
+        try:
+            t_1 = properties.t1(qubit)
+        except BackendPropertyError:
+            t_1 = None
+        try:
+            t_2 = properties.t2(qubit)
+        except BackendPropertyError:
+            t_2 = None
+        try:
+            frequency = properties.frequency(qubit)
+        except BackendPropertyError:
+            t_2 = None
+        qubit_props.append(
+            QubitProperties(  # type: ignore[no-untyped-call]
```

Comment:
Ok, at this point there's no way you're the only author of this.

Reply:
Heh, yeah I copied this code from the `qiskit-ibm-provider` which is based on an initial draft I wrote https://github.com/Qiskit/qiskit-ibm-provider/blob/main/qiskit_ibm_provider/utils/backend_converter.py and massaged a lot to pass mypy (which is run in the provider CI). But thanks for calling that out, I should add @rathishcholarajan and @kt474 to the co-authors list on this PR too since they did a bunch of work on the converter too.